//   Copyright 2011-2015 Quickstep Technologies LLC.
//   Copyright 2015 Pivotal Software, Inc.
//
//   Licensed under the Apache License, Version 2.0 (the "License");
//   you may not use this file except in compliance with the License.
//   You may obtain a copy of the License at
//
//       http://www.apache.org/licenses/LICENSE-2.0
//
//   Unless required by applicable law or agreed to in writing, software
//   distributed under the License is distributed on an "AS IS" BASIS,
//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//   See the License for the specific language governing permissions and
//   limitations under the License.

syntax = "proto2";

package quickstep.serialization;

import "storage/StorageBlockLayout.proto";

enum InsertDestinationType {
  ALWAYS_CREATE_BLOCK = 0;
  BLOCK_POOL = 1;
  PARTITION_AWARE = 2;
}

message InsertDestination {
  required InsertDestinationType insert_destination_type = 1;
  required int32 relation_id = 2;

  // Note(zuyu): false for UpdateOperator, and otherwise true.
  required bool need_to_add_blocks_from_relation = 3;

  // Note(zuyu): the actual StorageBlockLayout for insertion, if set.
  optional StorageBlockLayoutDescription layout = 4;

  required uint64 relational_op_index = 5;
}
